run_linux.go

```diff
@@ -1180,6 +1191,30 @@ func runCopyStdio(stdio *sync.WaitGroup, copyPipes bool, stdioPipe [][]int, copy
 				}()
 			}
 		}
+		//TOM: Start
+		if terminal.IsTerminal(unix.Stderr) {
+			if state, err := terminal.MakeRaw(unix.Stderr); err != nil {
+				logrus.Warnf("error setting terminal state: %v", err)
+			} else {
+				defer func() {
+					if err = terminal.Restore(unix.Stderr, state); err != nil {
+						logrus.Errorf("unable to restore terminal state: %v", err)
+					}
+				}()
+			}
+		}
+		if terminal.IsTerminal(unix.Stdout) {
+			if state, err := terminal.MakeRaw(unix.Stdout); err != nil {
+				logrus.Warnf("error setting terminal state: %v", err)
+			} else {
+				defer func() {
+					if err = terminal.Restore(unix.Stdout, state); err != nil {
+						logrus.Errorf("unable to restore terminal state: %v", err)
+					}
+				}()
+			}
+		}
+		//TOM: End
```

Comment:
@nalind, What I'm trying to do is to get the output of stderr from within the container to be returned back in `relayBuffer[unix.Stderr]`.  Currently and with these changes in play, it's always coming back in `relayBuffer[unix.Stdout]`.  I thought my changes above at 1170 would cure that, do you know if I'm off or I'm just not specifying them correctly?  Here's a snippet of the output I'm seeing with these changes in play:

```
# buildah run tom-working-container ls /fake
TOM: relayMap    All    [map[0:12 2:12 12:1]]
                                             TOM: relayBuffer stderr []
                                                                       TOM: relayBuffer stdout [ls: /fake: No such file or directory]
            TOM: Top drain output buffers
                                         ls: /fake: No such file or directory
```

Any sage advice?  If I can get stderr via the RelayBuffer, I could then color the output in theory, I just can't seem to get stderr there though.

Reply:
`relayMap` is used for deciding whiere data read from a given descriptor needs to go (which descriptor it needs to be written to), with the map keyed by the descriptor it's being read from.  `relayBuffer` is used for buffering data that hasn't been written to that destination yet, with the map keyed by the descriptor that it needs to be written to.  I wouldn't expect us to read from our own stderr descriptor, so setting `relayMap[unix.Stderr]` looks wrong.

The current logic routes the child process's stderr to our own stderr through `relayBuffer[unix.Stderr]`, but that only gets used when we have different descriptors to read the child's stdout and stderr data from (so, when we're using pipes).  When we're using a pseudoterminal, we only have one descriptor (`terminalFD`) from which to read both stdout and stderr combined, and we send all of the output to our stdout by queuing it up in `relayBuffer[unix.Stdout]`.

Reply:
@nalind thanks for the 411.  When I did `podman run $ctr /bin/sh` I was seeing the pipes in action with the separation.  You're right with the terminalFD in play, I wasn't seeing the separation.  So it sounds like the reason I couldn't find a solution while using the pseudoterminal, is there isn't a way to do so with that approach.  

Do you have any thoughts on how I might accomplish this, or is the risk of change to much for the gain of the coloring?

Reply:
To intercept only the contained process's stderr stream, we'd need to get between the process running in the container and the pseudoterminal that it's using for stdio, and preferably without interfering with its attempts to access the device using calls like `ioctl()` and `tcgetattr()`/`tcsetattr()`.  Using ptrace to do it would require capabilities that we don't usually have when we're run in a container, and I don't have a good idea of how much work would be required.